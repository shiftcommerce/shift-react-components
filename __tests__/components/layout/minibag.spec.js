// Libraries
import React from 'react'

// Components
import MiniBag from '../../../src/components/layout/minibag'

describe('Minibag', () => {
  test('renders the minibag links', () => {
    // arrange
    const cart = {
      line_items: [],
      line_items_count: 0
    }
    const className = 'dummy-classname'

    // act
    const wrapper = mount(
      <MiniBag cart={cart} className={className} />
    )

    // assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper).toIncludeText('Basket')
    expect(wrapper.find('.c-minibag__cart-image-count')).toIncludeText('0')
    expect(wrapper.find('.c-header__minibag')).toHaveClassName(className)
  })

  test('renders the line item quantity as expected, when line items are present', () => {
    // arrange
    const cart = {
      line_items: [
        {
          title: 'test',
          price: 10,
          discount: 0,
          quantity: 2,
          sku: '123',
          image_url: '',
          size: 'size - 8',
          productSku: '1231',
          productID: 1
        }
      ],
      line_items_count: 2
    }

    // act
    const wrapper = mount(
      <MiniBag cart={cart} />
    )

    // assert
    expect(wrapper.find('.c-minibag__cart-image-count')).toIncludeText('2')
  })

  test('renders an active checkout button when cart has items', () => {
    // arrange
    const cart = {
      line_items: [
        {
          title: 'test',
          price: 10,
          discount: 0,
          quantity: 2,
          sku: '123',
          image_url: '',
          size: 'size - 8',
          productSku: '1231',
          productID: 1
        }
      ],
      line_items_count: 2
    }

    // act
    const wrapper = mount(
      <MiniBag cart={cart} />
    )

    // assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.c-minibag__checkout .o-button a')).toIncludeText('Checkout')
    expect(wrapper.find('.c-minibag__checkout .o-button a')).not.toHaveClassName('o-button--disabled')
  })

  test('renders a disabled checkout button when cart is empty', () => {
    // arrange
    const cart = {
      line_items: [],
      line_items_count: 0
    }

    // act
    const wrapper = mount(
      <MiniBag cart={cart} />
    )

    // assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.c-minibag__checkout .o-button a')).toHaveClassName('o-button--disabled')
  })
})
