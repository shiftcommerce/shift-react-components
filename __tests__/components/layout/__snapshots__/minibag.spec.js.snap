// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Minibag doesnt render when miniBagDisplayed is false 1`] = `
<Minibag
  cart={
    Object {
      "line_items": Array [
        Object {
          "id": "311",
          "item": Object {
            "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg",
            "product": Object {
              "canonical_path": "/jugs/seed_product_40",
              "slug": "jugs/seed_product_40",
              "title": "First product",
            },
            "title": "First variant",
          },
          "item_id": 321,
          "item_type": "Variant",
          "line_item_discounts": Array [
            Object {
              "id": "2a720d8d-ae39-462e-b2e2-4c49ca5c8846",
              "line_item_number": 1,
              "promotion_id": 21,
              "total": 1.5,
            },
          ],
          "sku": "3008557600817",
          "stock_available_level": 85,
          "sub_total": 6.77,
          "total": 5.27,
          "total_discount": 1.5,
          "unit_quantity": 1,
        },
        Object {
          "id": "310",
          "item": Object {
            "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg",
            "product": Object {
              "canonical_path": "/jugs/seed_product_39",
              "slug": "jugs/seed_product_39",
              "title": "Second product",
            },
            "title": "Second variant",
          },
          "item_id": 321,
          "item_type": "Variant",
          "line_item_discounts": Array [
            Object {
              "id": "2a720d8d-ae39-462e-b2e2-4c49ca5c8846",
              "line_item_number": 1,
              "promotion_id": 21,
              "total": 1.5,
            },
          ],
          "sku": "3008557600817",
          "stock_available_level": 85,
          "sub_total": 6.77,
          "total": 5.27,
          "total_discount": 1.5,
          "unit_quantity": 1,
        },
      ],
      "line_items_count": 2,
    }
  }
  miniBagDisplayed={false}
/>
`;

exports[`Minibag doesnt render when there is no lineitems 1`] = `
<Minibag
  cart={
    Object {
      "line_items": Array [],
      "line_items_count": 0,
    }
  }
  className="dummy-classname"
  miniBagDisplayed={true}
>
  <div
    className="c-minibag__overlay"
    onClick={[Function]}
  />
  <div
    className="c-minibag__dropdown"
  >
    <div
      className="c-minibag__dropdown-container"
    >
      <section
        className="c-minibag__dropdown-header"
      >
        <h1
          className="c-minibag__dropdown-title"
        >
           Shopping Basket 
          <a
            className="c-checkout-cart__amount"
          >
            (
            0
            )
          </a>
        </h1>
        <input
          checked={true}
          className="c-minibag__dropdown-checkbox"
          id="minibag"
          readOnly={true}
          type="checkbox"
        />
        <label
          className="c-minibag__dropdown-cross"
          htmlFor="minibag"
          onClick={[Function]}
        />
      </section>
      <p>
        Your bag is empty.
      </p>
      <div
        className="c-minibag__dropdown-buttons"
      >
        <Link
          className="o-button o-button--sml o-button--primary c-minibag__dropdown-buttons--link"
          href="/cart"
          onClick={[Function]}
        >
          <a
            className="o-button o-button--sml o-button--primary c-minibag__dropdown-buttons--link"
            href="/cart"
            onClick={[Function]}
          >
            view shopping basket
          </a>
        </Link>
        <Button
          className="o-button--sml c-minibag__dropdown-buttons--link"
          label="continue shopping"
          onClick={[Function]}
          status="primary"
        >
          <button
            className="o-button--sml c-minibag__dropdown-buttons--link o-button o-button--primary"
            disabled={false}
            onClick={[Function]}
            role="button"
            tabIndex="0"
          >
            continue shopping
          </button>
        </Button>
      </div>
    </div>
  </div>
</Minibag>
`;

exports[`Minibag renders when miniBagDisplayed is true 1`] = `
<Minibag
  cart={
    Object {
      "discount_summaries": Array [],
      "line_items": Array [
        Object {
          "id": "310",
          "item": Object {
            "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg",
            "product": Object {
              "canonical_path": "/jugs/seed_product_39",
              "slug": "jugs/seed_product_39",
              "title": "Second product",
            },
            "sku": "3008557600818",
            "title": "Second variant",
          },
          "item_id": 321,
          "item_type": "Variant",
          "line_item_discounts": Array [
            Object {
              "id": "2a720d8d-ae39-462e-b2e2-4c49ca5c8846",
              "line_item_number": 1,
              "promotion_id": 21,
              "total": 1.5,
            },
          ],
          "stock_available_level": 85,
          "sub_total": 6.77,
          "total": 5.27,
          "total_discount": 1.5,
          "unit_quantity": 1,
        },
        Object {
          "id": "311",
          "item": Object {
            "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg",
            "product": Object {
              "canonical_path": "/jugs/seed_product_40",
              "slug": "jugs/seed_product_40",
              "title": "First product",
            },
            "sku": "3008557600817",
            "title": "First variant",
          },
          "item_id": 321,
          "item_type": "Variant",
          "line_item_discounts": Array [
            Object {
              "id": "2a720d8d-ae39-462e-b2e2-4c49ca5c8846",
              "line_item_number": 1,
              "promotion_id": 21,
              "total": 1.5,
            },
          ],
          "stock_available_level": 85,
          "sub_total": 6.77,
          "total": 5.27,
          "total_discount": 1.5,
          "unit_quantity": 1,
        },
      ],
      "line_items_count": 2,
    }
  }
  miniBagDisplayed={true}
  onItemQuantityUpdated={[MockFunction]}
>
  <div
    className="c-minibag__overlay"
    onClick={[Function]}
  />
  <div
    className="c-minibag__dropdown"
  >
    <div
      className="c-minibag__dropdown-container"
    >
      <section
        className="c-minibag__dropdown-header"
      >
        <h1
          className="c-minibag__dropdown-title"
        >
           Shopping Basket 
          <a
            className="c-checkout-cart__amount"
          >
            (
            2
            )
          </a>
        </h1>
        <input
          checked={true}
          className="c-minibag__dropdown-checkbox"
          id="minibag"
          readOnly={true}
          type="checkbox"
        />
        <label
          className="c-minibag__dropdown-cross"
          htmlFor="minibag"
          onClick={[Function]}
        />
      </section>
      <div
        className="c-minibag__line-items-section"
      >
        <div
          className="c-minibag__line-item"
          key="3008557600818"
        >
          <div
            className="c-minibag__line-item-information"
          >
            <div
              className="c-minibag__line-item-information-title"
            >
              <p>
                Second product - Second variant
              </p>
              <a
                className="c-minibag__line-item-total"
              >
                £
                5.27
              </a>
              <a
                className="c-minibag__line-item-subtotal"
              >
                £
                6.77
              </a>
            </div>
            <_class
              className="c-minibag__quantity"
              data-id="310"
              label="Quantity"
              onChange={[MockFunction]}
              options={
                Array [
                  Object {
                    "title": 1,
                    "value": 1,
                  },
                  Object {
                    "title": 2,
                    "value": 2,
                  },
                  Object {
                    "title": 3,
                    "value": 3,
                  },
                  Object {
                    "title": 4,
                    "value": 4,
                  },
                  Object {
                    "title": 5,
                    "value": 5,
                  },
                  Object {
                    "title": 6,
                    "value": 6,
                  },
                  Object {
                    "title": 7,
                    "value": 7,
                  },
                  Object {
                    "title": 8,
                    "value": 8,
                  },
                  Object {
                    "title": 9,
                    "value": 9,
                  },
                  Object {
                    "title": 10,
                    "value": 10,
                  },
                ]
              }
              skipPrompt={true}
              value={1}
            >
              <DropdownSelect
                className="c-minibag__quantity"
                data-id="310"
                label="Quantity"
                onChange={[MockFunction]}
                options={
                  Array [
                    Object {
                      "title": 1,
                      "value": 1,
                    },
                    Object {
                      "title": 2,
                      "value": 2,
                    },
                    Object {
                      "title": 3,
                      "value": 3,
                    },
                    Object {
                      "title": 4,
                      "value": 4,
                    },
                    Object {
                      "title": 5,
                      "value": 5,
                    },
                    Object {
                      "title": 6,
                      "value": 6,
                    },
                    Object {
                      "title": 7,
                      "value": 7,
                    },
                    Object {
                      "title": 8,
                      "value": 8,
                    },
                    Object {
                      "title": 9,
                      "value": 9,
                    },
                    Object {
                      "title": 10,
                      "value": 10,
                    },
                  ]
                }
                renderValidationMessage={[Function]}
                skipPrompt={true}
                value={1}
              >
                <div
                  className="o-form__input-group"
                >
                  <label
                    className="o-form__label"
                  >
                    <b>
                      Quantity
                    </b>
                  </label>
                  <select
                    className="o-form__input-block o-form__input-field c-minibag__quantity"
                    data-id="310"
                    label="Quantity"
                    onBlur={[Function]}
                    onChange={[Function]}
                    role="listbox"
                    value={1}
                  >
                    <option
                      aria-posinset={1}
                      aria-setsize={10}
                      key="0"
                      role="option"
                      value={1}
                    >
                      1
                    </option>
                    <option
                      aria-posinset={2}
                      aria-setsize={10}
                      key="1"
                      role="option"
                      value={2}
                    >
                      2
                    </option>
                    <option
                      aria-posinset={3}
                      aria-setsize={10}
                      key="2"
                      role="option"
                      value={3}
                    >
                      3
                    </option>
                    <option
                      aria-posinset={4}
                      aria-setsize={10}
                      key="3"
                      role="option"
                      value={4}
                    >
                      4
                    </option>
                    <option
                      aria-posinset={5}
                      aria-setsize={10}
                      key="4"
                      role="option"
                      value={5}
                    >
                      5
                    </option>
                    <option
                      aria-posinset={6}
                      aria-setsize={10}
                      key="5"
                      role="option"
                      value={6}
                    >
                      6
                    </option>
                    <option
                      aria-posinset={7}
                      aria-setsize={10}
                      key="6"
                      role="option"
                      value={7}
                    >
                      7
                    </option>
                    <option
                      aria-posinset={8}
                      aria-setsize={10}
                      key="7"
                      role="option"
                      value={8}
                    >
                      8
                    </option>
                    <option
                      aria-posinset={9}
                      aria-setsize={10}
                      key="8"
                      role="option"
                      value={9}
                    >
                      9
                    </option>
                    <option
                      aria-posinset={10}
                      aria-setsize={10}
                      key="9"
                      role="option"
                      value={10}
                    >
                      10
                    </option>
                  </select>
                </div>
              </DropdownSelect>
            </_class>
            <a
              className="c-minibag__line-item--delete"
              data-id="310"
            >
              Remove
            </a>
          </div>
          <div
            className="c-minibag__line-item-images"
          >
            <Link
              href="/slug?slug=/jugs/seed_product_39"
            >
              <a
                href="/slug?slug=/jugs/seed_product_39"
              >
                <Image
                  alt="Second variant"
                  aria-label="Second variant"
                  className="c-minibag__line-item-image"
                  key="jugs/seed_product_39"
                  src="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg"
                >
                  <div
                    className="c-minibag__line-item-image"
                  >
                    <picture>
                      <img
                        alt="Second variant"
                        aria-label="Second variant"
                        className="o-image"
                        src="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg"
                        srcSet="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg 769w"
                      />
                    </picture>
                  </div>
                </Image>
              </a>
            </Link>
          </div>
        </div>
        <div
          className="c-minibag__line-item"
          key="3008557600817"
        >
          <div
            className="c-minibag__line-item-information"
          >
            <div
              className="c-minibag__line-item-information-title"
            >
              <p>
                First product - First variant
              </p>
              <a
                className="c-minibag__line-item-total"
              >
                £
                5.27
              </a>
              <a
                className="c-minibag__line-item-subtotal"
              >
                £
                6.77
              </a>
            </div>
            <_class
              className="c-minibag__quantity"
              data-id="311"
              label="Quantity"
              onChange={[MockFunction]}
              options={
                Array [
                  Object {
                    "title": 1,
                    "value": 1,
                  },
                  Object {
                    "title": 2,
                    "value": 2,
                  },
                  Object {
                    "title": 3,
                    "value": 3,
                  },
                  Object {
                    "title": 4,
                    "value": 4,
                  },
                  Object {
                    "title": 5,
                    "value": 5,
                  },
                  Object {
                    "title": 6,
                    "value": 6,
                  },
                  Object {
                    "title": 7,
                    "value": 7,
                  },
                  Object {
                    "title": 8,
                    "value": 8,
                  },
                  Object {
                    "title": 9,
                    "value": 9,
                  },
                  Object {
                    "title": 10,
                    "value": 10,
                  },
                ]
              }
              skipPrompt={true}
              value={1}
            >
              <DropdownSelect
                className="c-minibag__quantity"
                data-id="311"
                label="Quantity"
                onChange={[MockFunction]}
                options={
                  Array [
                    Object {
                      "title": 1,
                      "value": 1,
                    },
                    Object {
                      "title": 2,
                      "value": 2,
                    },
                    Object {
                      "title": 3,
                      "value": 3,
                    },
                    Object {
                      "title": 4,
                      "value": 4,
                    },
                    Object {
                      "title": 5,
                      "value": 5,
                    },
                    Object {
                      "title": 6,
                      "value": 6,
                    },
                    Object {
                      "title": 7,
                      "value": 7,
                    },
                    Object {
                      "title": 8,
                      "value": 8,
                    },
                    Object {
                      "title": 9,
                      "value": 9,
                    },
                    Object {
                      "title": 10,
                      "value": 10,
                    },
                  ]
                }
                renderValidationMessage={[Function]}
                skipPrompt={true}
                value={1}
              >
                <div
                  className="o-form__input-group"
                >
                  <label
                    className="o-form__label"
                  >
                    <b>
                      Quantity
                    </b>
                  </label>
                  <select
                    className="o-form__input-block o-form__input-field c-minibag__quantity"
                    data-id="311"
                    label="Quantity"
                    onBlur={[Function]}
                    onChange={[Function]}
                    role="listbox"
                    value={1}
                  >
                    <option
                      aria-posinset={1}
                      aria-setsize={10}
                      key="0"
                      role="option"
                      value={1}
                    >
                      1
                    </option>
                    <option
                      aria-posinset={2}
                      aria-setsize={10}
                      key="1"
                      role="option"
                      value={2}
                    >
                      2
                    </option>
                    <option
                      aria-posinset={3}
                      aria-setsize={10}
                      key="2"
                      role="option"
                      value={3}
                    >
                      3
                    </option>
                    <option
                      aria-posinset={4}
                      aria-setsize={10}
                      key="3"
                      role="option"
                      value={4}
                    >
                      4
                    </option>
                    <option
                      aria-posinset={5}
                      aria-setsize={10}
                      key="4"
                      role="option"
                      value={5}
                    >
                      5
                    </option>
                    <option
                      aria-posinset={6}
                      aria-setsize={10}
                      key="5"
                      role="option"
                      value={6}
                    >
                      6
                    </option>
                    <option
                      aria-posinset={7}
                      aria-setsize={10}
                      key="6"
                      role="option"
                      value={7}
                    >
                      7
                    </option>
                    <option
                      aria-posinset={8}
                      aria-setsize={10}
                      key="7"
                      role="option"
                      value={8}
                    >
                      8
                    </option>
                    <option
                      aria-posinset={9}
                      aria-setsize={10}
                      key="8"
                      role="option"
                      value={9}
                    >
                      9
                    </option>
                    <option
                      aria-posinset={10}
                      aria-setsize={10}
                      key="9"
                      role="option"
                      value={10}
                    >
                      10
                    </option>
                  </select>
                </div>
              </DropdownSelect>
            </_class>
            <a
              className="c-minibag__line-item--delete"
              data-id="311"
            >
              Remove
            </a>
          </div>
          <div
            className="c-minibag__line-item-images"
          >
            <Link
              href="/slug?slug=/jugs/seed_product_40"
            >
              <a
                href="/slug?slug=/jugs/seed_product_40"
              >
                <Image
                  alt="First variant"
                  aria-label="First variant"
                  className="c-minibag__line-item-image"
                  key="jugs/seed_product_40"
                  src="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg"
                >
                  <div
                    className="c-minibag__line-item-image"
                  >
                    <picture>
                      <img
                        alt="First variant"
                        aria-label="First variant"
                        className="o-image"
                        src="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg"
                        srcSet="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg 769w"
                      />
                    </picture>
                  </div>
                </Image>
              </a>
            </Link>
          </div>
        </div>
      </div>
      <div
        className="c-minibag__dropdown-review"
      >
        <div
          className="c-minibag__dropdown-review-totals"
        >
          <div
            className="c-minibag__dropdown-review-total-line c-minibag__dropdown-review-total-line--main"
          >
            <p>
              Total:
            </p>
            <p>
              £
              NaN
            </p>
          </div>
        </div>
      </div>
      <div
        className="c-minibag__dropdown-buttons"
      >
        <Link
          className="o-button o-button--sml o-button--primary c-minibag__dropdown-buttons--link"
          href="/cart"
          onClick={[Function]}
        >
          <a
            className="o-button o-button--sml o-button--primary c-minibag__dropdown-buttons--link"
            href="/cart"
            onClick={[Function]}
          >
            view shopping basket
          </a>
        </Link>
        <Button
          className="o-button--sml c-minibag__dropdown-buttons--link"
          label="continue shopping"
          onClick={[Function]}
          status="primary"
        >
          <button
            className="o-button--sml c-minibag__dropdown-buttons--link o-button o-button--primary"
            disabled={false}
            onClick={[Function]}
            role="button"
            tabIndex="0"
          >
            continue shopping
          </button>
        </Button>
      </div>
    </div>
  </div>
</Minibag>
`;
